pyam/core.py

```diff
@@ -2793,6 +2793,8 @@ def concat(objs, ignore_meta_conflict=False, **kwargs):
     The *meta* attributes are merged only for those objects of *objs* that are passed
     as :class:`IamDataFrame` instances.
 
+    The index dimension of the return value is carried over from index of the *meta*
+    indicators.
```

Comment:
Add an explanation that index and data dimensions have to be identical.
```suggestion
    The :attr:`dimensions` and :attr:`index` names of all elements of *dfs* must be
    identical. The returned IamDataFrame inherits the dimensions and index names.
```